File: challenges/5-cobaias/elixir/digzom/cobaias.exs
defmodule Cobaias do
  def calculate_amount(data) do
    data
    |> Enum.map(&take_quantity/1)
    |> Enum.reduce(fn qnt, sum -> qnt + sum end)
  end

  defp format_data(data) do
    data
    |> String.split("\n")
    |> Enum.drop(1)
  end

  def take_quantity(string) do
    [quantity, _] = String.split(string)

    String.to_integer(quantity)
  end

  def animal_amount(data, letter) do
    data
    |> Enum.filter(fn str -> subject(str) === letter end)
    |> calculate_amount()
  end

  def subject(string) do
    string |> String.at(-1)
  end

  def calculate_percentage(total_amount, animal_amount) do
    (animal_amount / total_amount * 100) |> :erlang.float_to_binary(decimals: 2)
  end

  def present_data(data) do
    dropped_experiment_total = format_data(data)

    amount = calculate_amount(dropped_experiment_total)
    rabbit_amount = animal_amount(dropped_experiment_total, "C")
    frogg_amount = animal_amount(dropped_experiment_total, "S")
    rat_amount = animal_amount(dropped_experiment_total, "R")

    rabbit_percentage = calculate_percentage(amount, rabbit_amount)
    rat_percentage = calculate_percentage(amount, rat_amount)
    frogg_percentage = calculate_percentage(amount, frogg_amount)

    IO.puts(~s(
    Total: #{amount} cobaias
    Total de coelhos: #{rabbit_amount}
    Total de ratos: #{rat_amount}
    Total de sapos: #{frogg_amount}
    Percentual de coelhos: #{rabbit_percentage} %
    Percentual de ratos: #{rat_percentage} %
    Percentual de sapos: #{frogg_percentage} %
    ))
  end
end

System.argv()
|> File.read!()
|> Cobaias.present_data()

File: challenges/5-cobaias/python/uKaigo/cobaias.py
nexp = int(input('Insira o número de experimentos: '))

animals = {'c': 0, 'r': 0, 's': 0}

for i in range(1, nexp + 1):
    count, anim_typ = input(f'{i}º: ').lower().split()

    animals[anim_typ[0]] += int(count)

tot_anim = sum(animals.values())

print(f'Total: {tot_anim} cobaias')
print(f'Total de coelhos: {animals["c"]}')
print(f'Total de ratos: {animals["r"]}')
print(f'Total de sapos: {animals["s"]}')

print(f'Percentual de coelhos: {animals["c"]/tot_anim:.2%}')
print(f'Percentual de ratos: {animals["r"]/tot_anim:.2%}')
print(f'Percentual de sapos: {animals["s"]/tot_anim:.2%}')
